official/resnet/keras/keras_imagenet_main.py

```diff
@@ -0,0 +1,237 @@
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import time
+
+from absl import app as absl_app
+from absl import flags
+import tensorflow as tf  # pylint: disable=g-bad-import-order
+
+from official.resnet import imagenet_main
+from official.resnet import imagenet_preprocessing
+from official.resnet import resnet_run_loop
+from official.resnet.keras import resnet_model
+from official.utils.flags import core as flags_core
+from official.utils.logs import logger
+from official.utils.misc import distribution_utils
+
+
+class TimeHistory(tf.keras.callbacks.Callback):
+  """Callback for Keras models."""
+
+  def __init__(self, batch_size):
+    """Callback for Keras models.
+
+    Args:
+      batch_size: Total batch size.
+
+    """
+    self._batch_size = batch_size
+    super(TimeHistory, self).__init__()
+
+  def on_train_begin(self, logs=None):
+    self.epoch_times = []
+    self.batch_times = []
+    self.record_batch = True
+
+  def on_epoch_begin(self, epoch, logs=None):
+    self.epoch_time_start = time.time()
+
+  def on_epoch_end(self, epoch, logs=None):
+    self.epoch_times.append(time.time() - self.epoch_time_start)
+
+  def on_batch_begin(self, batch, logs=None):
+    if self.record_batch:
+      self.batch_time_start = time.time()
+      self.record_batch = False
+
+  def on_batch_end(self, batch, logs=None):
+    if batch % 100 == 0:
+      last_100_batches = time.time() - self.batch_time_start
+      examples_per_second = (self._batch_size * 100) / last_100_batches
+      self.batch_times.append(last_100_batches)
+      self.record_batch = True
+      # TODO(anjalisridhar): add timestamp as well.
+      if batch != 0:
+        tf.logging.info("BenchmarkMetric: {'num_batches':%d, 'time_taken': %f,"
+                        "'images_per_second': %f}" %
+                        (batch, last_100_batches, examples_per_second))
+
+
+def synthetic_input_fn(batch_size, height, width, num_channels, num_classes,
+                       dtype=tf.float32):
+  """Returns dataset filled with random data."""
+  # Synthetic input should be within [0, 255].
+  inputs = tf.truncated_normal(
+      [batch_size] + [height, width, num_channels],
+      dtype=dtype,
+      mean=127,
+      stddev=60,
+      name='synthetic_inputs')
+
+  labels = tf.random_uniform(
+      [batch_size],
+      minval=0,
+      maxval=num_classes - 1,
+      dtype=tf.int32,
+      name='synthetic_labels')
+
+  dataset = tf.data.Dataset.from_tensors((inputs, labels)).repeat()
+  dataset = dataset.prefetch(buffer_size=tf.contrib.data.AUTOTUNE)
+  return dataset
+
+
+def run_imagenet_with_keras(flags_obj):
+  """Run ResNet ImageNet training and eval loop using native Keras APIs.
+
+  Args:
+    flags_obj: An object containing parsed flag values.
+
+  Raises:
+    ValueError: If fp16 is passed as it is not currently supported.
+  """
+  if flags_obj.dtype == 'fp16':
```

Comment:
How about:

```python
dtype = flags_core.get_tf_dtype(flags_obj)
if dtype == tf.float15:
  raise ValueError(...)
```

And then we can use this same `dtype` in the calls to `synthentic_input_fn` calls below.
This will make it clearer that the 3 uses of `dtype` are all the same (instead of using the `.dtype` property here and then the `flags_core` function further on)